lib/utils.pm

```diff
@@ -460,6 +476,18 @@ sub reboot_gnome {
     workaround_type_encrypted_passphrase;
 }
 
+=head2 reboot
+
+    reboot;
+
+Reboot system from root console.
+=cut
+sub reboot {
+    select_console 'root-console';
+    type_string "reboot\n";
+    xen_restore_system;
```

Comment:
I usually do it like you suggest, but this time I put the `if (check_var('VIRSH_VMM_FAMILY', 'xen'))` in `xen_restore_system()` as `return if (check_var('VIRSH_VMM_FAMILY', 'xen'))`. Someone else having opinion on this?